pkg/client/unversioned/helper.go

```diff
@@ -408,7 +421,16 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 		return nil, err
 	}
 
-	client := NewRESTClient(baseURL, config.Version, config.Codec, config.QPS, config.Burst)
+	// baseURL is versioned, let's get a base without the version
+	configCopy := *config
+	configCopy.Version = ""
+	unversioned, err := defaultServerUrlFor(&configCopy)
```

Comment:
This kind of code repeats multiple times. Maybe it's worth to factor that out somehow?